Move module-level comment out of _RNNCell class

choucavalier opened the pull request:
The _RNNCell class docstring contained a description of the module
itself. I moved it back to the module's docstring.
I also fixed a comment that wasn't correctly indented.

choucavalier committed:
Move module-level comment out of _RNNCell class

The _RNNCell class docstring contained a description of the module
itself. I moved it back to the module's docstring.
I also fixed a comment that wasn't correctly indented.

tensorflow-jenkins commented:
Can one of the admins verify this patch?

jhseu commented:
Perhaps the comment belongs in `tensorflow/contrib/rnn/python/ops/core_rnn_cell_impl.py` instead?

tgy commented:
@jhseu what's the rationale behind the existence of both `tensorflow/python/ops/rnn_cell_impl.py` and `tensorflow/contrib/rnn/python/ops/core_rnn_cell_impl.py`? why not merge them in a single file?

jhseu commented:
contrib specifies APIs that aren't stable yet. Some parts of the RNN API have stabilized, and others haven't.

martinwicke commented on tensorflow/python/ops/rnn_cell_impl.py:
```diff
@@ -13,7 +13,15 @@
+operators that allow adding dropouts, projections, or embeddings for inputs.
+Constructing multi-layer cells is supported by the class `MultiRNNCell`, or by
+calling the `rnn` ops several times. Every `RNNCell` must have the properties
+below and implement `__call__` with the following signature.
```
The text isn't quite right for a module docstring, and not quite right for a class docstring. Can you disentangle the two?

tgy replied:
Done.

choucavalier committed:
Reorganizing rnn_cell_impl module docstrings

- Moves the module docstring of `python.ops.rnn_cell_impl` to
`contrib.rnn.python.ops.core_rnn_cell_impl`, where the actual
implementation of all the LSTM cells is located.
- Adds a module docstring to `python.ops.rnn_cell_impl`, explaining
that the module contains an abstract definition of the `RNNCell` and
that the actual implementations are located in `contrib`.

martinwicke commented:
Jenkins, test this please.

tgy commented:
Great! Maybe close this?

martinwicke commented:
It is merged already.
